lib/stack/src/watch.rs

```diff
@@ -14,14 +14,16 @@ pub trait WithUpdate<U> {
 }
 
 #[derive(Debug)]
-pub struct Layer<U> {
+pub struct Layer<T: WithUpdate<U>, U, M> {
     watch: Watch<U>,
+    _p: ::std::marker::PhantomData<(T, M)>,
```

Comment:
Nit: this should probably be `PhantomData<fn() -> (T, M)>` in case `T` or `M` are not `Send`/`Sync`?

Reply:
Alternatively, TIOLI, I think `Layer` could just be
```rust
pub struct Layer<U> {
    watch: Watch<U>
}
```
and the `T` and `M` types could only be present on the functions that need the `T: WithUpdate<U>` bound...

Reply:
it doesn't work out that way in practice. PhantomData is necessary.